src/utils/timeTracker/index.js

```diff
@@ -0,0 +1,188 @@
+      tracker.count && tracker.count(resp.status);
+
+      return resp;
+    })
+    .catch(err => {
+      this.stop(task, modifier);
+
+      const tracker = CALLBACKS[task] && CALLBACKS[task].metrics();
+      tracker.countException && tracker.countException();
+
+      throw err;
+    });
+  },
+  /**
+   * Starts tracking the time for a given task.
+   *
+   * @param {string} task - The task we are starting.
+   * @param {Promise} promise - (optional) The promise we are tracking.
+   */
+  start(task, promise) {
+    let result;
+    // If we are registering a promise with this task, we should count the status and the exceptions as well
+    // as stopping the task when the promise resolves.
+    if (thenable(promise)) {
+      result = this.__attachToPromise(promise, task);
+    }
+
+    // Start the timer, then save the reference. We do it last to avoid counting as much extra processing
+    // as possible on the latencies.
+    timers[task] = timer();
+
+    // If no promise is present, we will return undefined as before.
+    return result;
+  },
+  /**
+   * Starts tracking the time for a given UNIQUE task. By unique task we mean tasks that fall under the same concept
+   * but each call should be tracked in a unique way. (So if you start two timers for the same task before stopping,
+   * we should track them separately)
+   *
+   * @param {string} task - The task we are starting.
+   * @return {Function} The stop function for this specific task.
+   */
+  startUnique(task, promise) {
+    const taskUniqueId = uniqueId();
+    let result;
+
+    // If we are registering a promise with this task, we should count the status and the exceptions as well
+    // as stopping the task when the promise resolves. Then return the promise
+    if (thenable(promise)) {
+      result = this.__attachToPromise(promise, task, taskUniqueId);
+    } else {
+      // If not, we return the stop function, as it will be stopped manually.
+      result = this.stop.bind(this, task, taskUniqueId);
+    }
+
+    // We start the timer, with an uniqueId attached to it's name.
+    timers[task + taskUniqueId] = timer();
+
+    return result;
+  },
+  /**
+   * Setup the metricTrackers variable for this module. This method should be called as soon as
+   * the metrics get instantiated.
+   *
+   * @param {Object} trackers - The object containing all the metrics tracker modules.
+   */
+  setupTrackers(trackers) {
+    metricTrackers = trackers;
+  },
+  /**
+   * Stops the tracking of a given task.
+   *
+   * @param {string} task - The task we are starting.
+   * @param {string} modifier - (optional) The modifier for that specific task.
+   */
+  stop(task, modifier) {
+    const timerName = typeof modifier === 'string' ? task + modifier : task;
+    const timer = timers[timerName];
+    if (timer) {
+      // Stop the timer and round result for readability.
+      const et = Math.round( timer() );
```

Comment:
I guess that I can move it inside the timer. It was outside because the `timer` utility was meant to be generic. But if we reuse it somewhere else (since this changes the only one using it is the timeTracker) I guess that the decimals won't be important, and we can always change.

I'll move it to the timer.